From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: laurenceputterman <mlputterman@gmail.com>
Date: Sun, 17 May 2026 11:33:10 -0700
Subject: [PATCH 3/9] android netplay thread priority

---
 Source/Core/Core/Slippi/SlippiNetplay.cpp | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/Source/Core/Core/Slippi/SlippiNetplay.cpp b/Source/Core/Core/Slippi/SlippiNetplay.cpp
index 0232489d6b33c214042f23b3a11a30d6f9e8a0d7..f84da0d4a2c033d1628d4ee9d84d3895a0b0b8e4 100644
--- a/Source/Core/Core/Slippi/SlippiNetplay.cpp
+++ b/Source/Core/Core/Slippi/SlippiNetplay.cpp
@@ -7,12 +7,16 @@
 #include <fstream>
 #include <memory>
 #include <thread>
+#ifdef __ANDROID__
+#include <sys/resource.h>
+#endif
 
 #include "Common/CommonTypes.h"
 #include "Common/Config/Config.h"
 #include "Common/ENet.h"
 #include "Common/Logging/Log.h"
 #include "Common/MsgHandler.h"
+#include "Common/Thread.h"
 #include "Common/Timer.h"
 #include "Core/Config/GraphicsSettings.h"
 #include "Core/Config/NetplaySettings.h"
@@ -784,6 +788,11 @@ void SlippiNetplayClient::SendAsync(std::unique_ptr<sf::Packet> packet)
 // called from ---NETPLAY--- thread
 void SlippiNetplayClient::ThreadFunc()
 {
+  Common::SetCurrentThreadName("NetPlay Client");
+#ifdef __ANDROID__
+  setpriority(PRIO_PROCESS, 0, -8);
+#endif
+
   // Let client die 1 second before host such that after a swap, the client won't be connected to
   u64 start_time = Common::Timer::NowMs();
   u64 timeout = 8000;
